### PR TITLE
chore(medusa): Improve database loader error handling

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CORE-1400

### CODE REVIEW

**Review by olivermrbl — COMMENTED** [review]
Should we place this in `utils` to make it reusable in our modules?

File: packages/utils/src/common/handle-postgres-database-error.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { EOL } from "os"
```
Comment (olivermrbl) [inline packages/utils/src/common/handle-postgres-database-error.ts:1]: praise: Very nice!

File: packages/utils/src/common/handle-postgres-database-error.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  if (DatabaseErrorCode.connectionFailure === err.code) {
+    throw new Error(
+      `The connection cannot be established to the specified PostgreSQL database. Please, check the following point.
+      - Is the database running?
+      - Is the database url correct?
+      - Is the database port correct?
+      Please, also verify that your medusa-config.js database_url is correctly formatted. It should be in the following format:
+      postgres://user:password@host:post/db_name - If there is no password, you can omit it.
```
Comment (olivermrbl) [inline packages/utils/src/common/handle-postgres-database-error.ts]:
```suggestion
      `Failed to establish a connection to PostgreSQL. Please ensure the following is true and try again:
      - You have a PostgreSQL database running
      - You have passed the correct credentials in medusa-config.js
      - You have formatted the database connection string correctly. See below:
      "postgres://[username]:[password]@[host]:[post]/[db_name]" - If there is no password, you can omit it from the connection string
```

File: packages/utils/src/common/handle-postgres-database-error.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (DatabaseErrorCode.wrongCredentials === err.code) {
+    throw new Error(
+      `The specified credentials does not exists for the specified PostgreSQL database.${EOL}${err.message}`
+    )
+  }
+
+  if (DatabaseErrorCode.notFound === err.code) {
+    throw new Error(
+      `The specified connection url for your PostgreSQL database might contains illegal characters. Please check that all the connection segments contains only allowed characters [a-zA-Z0-9]${EOL}${err.message}`
```
Comment (olivermrbl) [inline packages/utils/src/common/handle-postgres-database-error.ts]:
```suggestion
      `The specified connection string for your PostgreSQL database might have illegal characters. Please check that it only contains allowed characters [a-zA-Z0-9]${EOL}${err.message}`
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]
love this!

**Review by pKorsholm — APPROVED** [review]
LGTM w/ Oli's changes 😄

**Comment by olivermrbl** [comment]
Unit tests need updating 👍

**Pull request merged**